src/services/mongoose/users.ts

```diff
@@ -88,6 +86,36 @@ export const UsersRepository = (): IUsersRepository => {
     }
   }
 
+  const migrateUserIdSubject = async ({
+    currentUserIdSubject,
+    newUserIdSubject,
+    phone,
+  }: {
+    currentUserIdSubject: UserId
+    newUserIdSubject: UserId
+    phone: PhoneNumber
+  }): Promise<User | RepositoryError> => {
+    const filter = { userId: { $eq: currentUserIdSubject } }
+    const update = {
+      $set: {
+        userId: newUserIdSubject,
+        phone,
+        deviceId: currentUserIdSubject,
```

Comment:
could be even easier to just set it at the time of id creation?

I guess it'll lead to some data duplicate, but the pros is that one query would be able to give us all deviceId, otherwise we need to do some join to get all the deviceId with current scheme.